subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/Cancellable.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.execution;
+
+/**
+ * Indicates that a task is cancellable.
+ */
+public interface Cancellable {
+    /**
+     * When Ctrl-C is pressed during the task execution, this method of that task will be invoked.
```

Comment:
❌ Builds can be cancelled in other ways. This should not get into the detail how the build was cancelled.

Reply:
Done. Thanks.